compiler/src/dotty/tools/dotc/transform/TreeChecker.scala

```diff
@@ -306,6 +306,7 @@ class TreeChecker extends Phase with SymTransformer {
       sym.isEffectivelyErased && sym.is(Private) && !sym.initial.is(Private)
 
     override def typed(tree: untpd.Tree, pt: Type = WildcardType)(using Context): Tree = {
+      TreeNodeChecker.run(tree)
```

Comment:
`typed` is run on every tree node and `TreeNodeChecker` again traverses its input, so this accidentally makes the TreeChecker quadratic

Reply:
That mirrors `checkIdentNotJavaClass`.  What can be used instead?